src/llmcompressor/transformers/finetune/runner.py

```diff
@@ -251,21 +260,30 @@ def run_sequential_stages(self, checkpoint: Optional[str] = None):
 
             # run stage
             if run_type is StageRunType.ONESHOT:
-                self.one_shot(stage=stage_name)
+                from llmcompressor.transformers.calibration import Oneshot
+
+                model = get_session_model()
+                self._model_args.model = model
+
+                oneshot = Oneshot(
+                    output_dir=self._training_args.output_dir,
+                    **get_dataclass_as_dict(self._model_args, ModelArguments),
```

Comment:
This is a really weird pattern, where we parse the data args, then expand them to kwargs, then parse them again within Oneshot.

This will be simpler in the future if we decide to make Oneshot (or LLMCompressor) the main entrypoint for user args, but in the meantime we can add a Oneshot.from_args class method that just assigns the args directly?

Reply:
ok `Oneshot.from_args` is a good idea